core/controllers/editor.py

```diff
@@ -775,3 +777,53 @@ def get(self, exploration_id):
         self.render_json({
             'unresolved_answers': unresolved_answers_with_frequency
         })
+
+
+class LearnerAnswerInfoHandler(EditorHandler):
+    """Handles the learner answer info for an exploration state."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_edit_exploration
+    def get(self, exploration_id):
+        """Handles the GET requests for for learner answer info for an
+        exploration state.
+        """
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        state_name = self.request.get('state_name')
+        if not state_name:
+            raise self.InvalidInputException
+
+        entity_type = feconf.ENTITY_TYPE_EXPLORATION
+        state_reference = (
+            stats_models.LearnerAnswerDetailsModel
+            .get_state_reference_for_exploration(exploration_id, state_name))
+        learner_answer_details = stats_services.get_learner_answer_details(
+            entity_type, state_reference)
+        learner_answer_info_dict_list = []
+        if learner_answer_details is not None:
+            learner_answer_info_dict_list = [
+                learner_answer_info.to_dict() for learner_answer_info in
+                learner_answer_details.learner_answer_info_list]
+        self.render_json({
```

Comment:
What if a user provides invalid state_name? Are we going to throw any error?

Reply:
If the state created by the editor is invalid then no learner answer details will be picked from the backend.

Reply:
But do we expect to have a valid state_name from the frontend?

Reply:
@anubhavsinha98 You need to assume that the frontend can pass the backend _anything_, and validate accordingly. Please make sure you are handling all the cases.

Reply:
I have added the validation for both question and exploration